arangod/Cluster/ClusterInfo.cpp

```diff
@@ -122,6 +122,13 @@ struct ShardStatistics {
 
 namespace {
 
+struct CreateCollectionReport {
+  std::optional<ErrorCode> dbServerResult{std::nullopt};
+  size_t nrDone{0};
+  std::string errMsg{StaticStrings::Empty};
```

Comment:
Why not combine `dbServerResult` and `errMsg` into an `arangodb::Result`?

Reply:
Also it's not necessary to init `std::string` by `StaticStrings::Empty`, it's empty by default

Reply:
I agree this would be better.
I just wanted to keep the change as small as possible and tried to "just" move the lock-protected entries into one data-structure.
Given that we most likely want to backport this change.

Will try locally how much I need to adapt for this change.

Reply:
Applied change